File: conway-ruby/lib/conway/game_presenter.rb
module Conway
  class GamePresenter
    def initialize(game_model)
      @game_model = game_model
    end

    def print_pattern
      system 'clear'
      puts('Board')
      board_string = ''
      (@game_model.start_row - 1..@game_model.end_row).each do |row|
        (@game_model.start_col - 1..@game_model.end_col).each do |col|
          board_string += @game_model.grid_state[row][col]
        end
        board_string += "\n"
      end

      puts(board_string)
    end
  end
end

File: conway-java/src/main/java/com/gojek/pasti/conway/GamePresenter.java
package com.gojek.pasti.conway;

public class GamePresenter {
  private GameModel gameModel;

  public GamePresenter(GameModel gameModel) {
    this.gameModel = gameModel;
  }

  public void printPattern() {
    clearScreen();
    System.out.println("Board");
    for (int row = gameModel.startRow - 1; row <= gameModel.endRow; row++) {
      for (int col = gameModel.startCol - 1; col <= gameModel.endCol; col++) {
        System.out.print(gameModel.gridState[row][col]);
      }
      System.out.print("\n");
    }
  }

  private void clearScreen() {
    System.out.print("\033[H\033[2J");
  }
}
